### PR TITLE
added Makefile example for compiling,linking apps to use Data Broker

### PR DESCRIPTION
**Pull request by damora**
added section showing an example Makefile for application developers trying to use data broker

### CODE REVIEW

File: doc/guide.tex [hunk 1]
```diff
@@ -416,8 +416,46 @@ \subsection{Example Sequence}
...
+all: Makefile $(TARGET)
+$(TARGET): $(OBJECTS) Makefile
+    $(LINK) $(LDFLAGS) -o $(TARGET) $(OBJECTS) $(LIBS)
+# 
+clean:
+    $(RM)  $(OBJECTS) $(DESTDIR)/$(TARGET)
+\end{lstlisting}
 
-\subsection{Configuration}
+\subsection{Configuration}\
```
Comment (lasch) [inline doc/guide.tex]: Accidental \ in there?

File: doc/guide.tex [hunk 2]
```diff
@@ -416,8 +416,46 @@ \subsection{Example Sequence}
...
+\begin{lstlisting}[basicstyle=\scriptsize\ttfamily,backgroundcolor=\color{backcolour},caption=Example Makefile,captionpos=b,label=code:Makefile]
+# Build Environment
+CC            = gcc
+CXX           = g++
+CFLAGS        = -O3
+CXXFLAGS      = -O3
+INCPATH       = -I./ -I$(DATABROKER_HOME)/include
+LINK          = g++ 
+LDFLAGS       = -L$(DATABROKER_HOME)/lib
+LIBS          = -lcurses -lstdc++ -ldatabroker -ldbbe_redis
```
Comment (lasch) [inline doc/guide.tex]: linking libcurses and libstdc++ in general?

**Review by lasch — COMMENTED** [review]
Thanks for putting this together.
Looks good to me. Just one minor item to check as commented.

File: doc/guide.tex (hunk 1 above)
Comment (damora) [reply doc/guide.tex]: Perhaps, but agree it does not belong

**Comment by damora** [comment]
removed / after Configuration and also removed -lcurses. I think -lstdc++ should remain

**Review by lasch — APPROVED** [review]
The stdc++ would stay for this case as it's building/linking a C++ source/example. I got confused by the generic .C.o: target. So for a pure C example, the LIBS variable would have to be adjusted.

**Pull request merged**